Apply the equivalent of:

#commit 32bfd35d4b63bd63de4bb0d791ef049c3c868726
#Author: Johannes Berg <johannes@sipsolutions.net>
#Date:   Wed Dec 19 01:31:26 2007 +0100
#
#    mac80211: dont use interface indices in drivers
#
#    This patch gets rid of the if_id stuff where possible in favour of
#    a new per-virtual-interface structure "struct ieee80211_vif". This
#    structure is located at the end of the per-interface structure and
#    contains a variable length driver-use data area.
#
#    This has two advantages:
#     * removes the need to look up interfaces by if_id, this is better
#       for working with network namespaces and performance
#     * allows drivers to store and retrieve per-interface data without
#       having to allocate own lists/hash tables
#
#    Signed-off-by: Johannes Berg <johannes@sipsolutions.net>
#    Signed-off-by: John W. Linville <linville@tuxdriver.com>

Changes-licensed-under: ISC
Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/drivers/net/wireless/ath/ar5523/ar5523.c
+++ b/drivers/net/wireless/ath/ar5523/ar5523.c
@@ -966,7 +966,8 @@ static int ar5523_config(struct ieee8021
 	return 0;
 }
 
-static int ar5523_config_interface(struct ieee80211_hw *hw, int if_id,
+static int ar5523_config_interface(struct ieee80211_hw *hw,
+		struct ieee80211_vif *vif,
 		struct ieee80211_if_conf *ifconf)
 {
 	struct ar5523 *ar = hw->priv;
